jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/Validator.java

```diff
@@ -27,35 +28,120 @@
-        !value.trim().isEmpty(), "Property '" + propertyName + "' cannot be empty");
+        !value.trim().isEmpty(), "Property '" + propertyName + "' cannot be an empty string");
   }
 
   /**
-   * Checks if a collection is null or empty.
+   * Checks if string is null or non empty.
+   *
+   * @param value the string in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code value} is empty or only whitespace
+   */
+  public static void checkNullOrNotEmpty(@Nullable String value, String propertyName) {
+    if (value == null) {
+      // pass
+      return;
+    }
+    Preconditions.checkArgument(
+        !value.trim().isEmpty(), "Property '" + propertyName + "' cannot be an empty string");
+  }
+
+  /**
+   * Checks if a collection is not null and not empty.
    *
    * @param value the string in question
    * @param propertyName the equivalent 'yaml' property name
    * @throws NullPointerException if {@code value} is null
    * @throws IllegalArgumentException if {@code value} is empty
    */
-  public static void checkNotEmpty(@Nullable Collection<?> value, String propertyName) {
+  public static void checkNotNullAndNotEmpty(@Nullable Collection<?> value, String propertyName) {
     Preconditions.checkNotNull(value, "Property '" + propertyName + "' cannot be null");
     Preconditions.checkArgument(
         !value.isEmpty(), "Property '" + propertyName + "' cannot be an empty collection");
   }
 
   /**
-   * Checks if string is what is expected.
+   * Check if a collection is either null, empty or contains only non-null, non-empty values.
+   *
+   * @param values the collection in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code values} contains empty entries
+   * @throws NullPointerException if {@code values} contains null entries
+   */
+  public static void checkNullOrNonNullNonEmptyEntries(
+      @Nullable Collection<String> values, String propertyName) {
+    if (values == null || values.isEmpty()) {
+      // pass
+      return;
+    }
+    for (String value : values) {
+      Preconditions.checkNotNull(
+          value, "Property '" + propertyName + "' cannot contain null entries");
+      Preconditions.checkArgument(
+          !value.trim().isEmpty(), "Property '" + propertyName + "' cannot contain empty strings");
+    }
+  }
+
+  /**
+   * Check if a map is either null, empty or contains only non-null, non-empty keys and values.
+   *
+   * @param values the collection in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code values} contains empty keys or values
+   * @throws NullPointerException if {@code values} contains null keys or values
+   */
+  public static void checkNullOrNonNullNonEmptyEntries(
+      @Nullable Map<String, String> values, String propertyName) {
+    if (values == null || values.isEmpty()) {
```

Comment:
Nit: remove `values.isEmpt()` for consistency?

Reply:
oh yeah good catch.